### DISCUSSION TITLE
How can I have Meson reconize custom media files for testing?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13623] [Q&A]
Building a database system and would like to have the test use the example files in the test data directory, how can I do this with Meson 1.3.0 and newer?

Build file for reference, located in the test directory along with sub directories for test data and scripts:
https://github.com/dreamer-coding/fossil-crabdb/blob/crabdb_prod/code/tests/meson.build

In addition to setting up test data I'm also wondering how would I use media files in a given Meson build project that would make use of database files and other types of media.

**Comment by dreamer-coding** [comment]
This seems to work so far:
```meson
fs = import('fs')
data = [
    'delete_command.crabql',
    'insert_commands.crabql',
    'select_command.crabql',
    'sort_key_command.crabql',
    'test_condition_met.crabql',
    'test_condition_not_met.crabql',
    'test_invalid.txt',
    'test_unknown_command.crabql',
    'test_valid.crabql',
    'update_command.crabql'
]

foreach iter : data
    fs.copyfile(iter, iter)
endforeach
```

**Reply by dreamer-coding** [reply]
It seems to fail, still trying not to use `configure_file` as it seems to not be recommended for something such as making the project aware of media files

**Comment by dcbaker** [comment]
I'm assuming that your test programs expect to find test files in a certain directory, like the one it's run in (the link you gave provides a 404)?

**Reply by dreamer-coding** [reply]
Sorry about that, I have been busy, yeah that's the idea: here is a stable link, still something I need to figure out: https://github.com/fossillogic/fossil-crabdb/tree/main/code/tests

**Reply by dreamer-coding** [reply]
> Regarding the project media how can I make it work for both test and application? https://github.com/dreamer-coding/fossil-project/blob/main/code/media/meson.build

**Comment by eli-schwartz** [comment]
You haven't really explained what the problem is. You described something you have tried so far, but then say "It seems to fail" without a description of what, precisely, failed.

**Reply by dreamer-coding** [reply]
```
Building a database system and would like to have the test use the example files in the test data
directory, how can I do this with Meson 1.3.0 and newer?

Build file for reference, located in the test directory along with sub directories for test data and scripts:
https://github.com/dreamer-coding/fossil-crabdb/blob/crabdb_prod/code/tests/meson.build

In addition to setting up test data I'm also wondering how would I use media files in a given Meson build project that would make use of database files and other types of media.
```

**Reply by eli-schwartz** [reply]
This reply doesn't contain a description of how your current attempt has failed. All you did was repeat the original question.

**Reply by eli-schwartz** [reply]
Also please keep in mind that "media" and "databases" are attributes of the file contents and how you intend to use them.

Nodes on a build graph don't consider the contents of the file as part of the graph resolution algorithm. The fact that they are databases or "media" is not a load-bearing part of the question and should be omitted for brevity.
